### ISSUE TITLE
RP does not pull from the database

### ISSUE THREAD

**Issue by iparask** [radical-cybertools/radical.pilot#1342]
Based on  a discussion I had with Andre, I changed some configurations of the agent like how many CUs should be in the DB before starting to pull and how often to poll the database. Although all units were submitted the agent did not pull anything from the database.

My stack is:
```
python            : 2.7.6
virtualenv        : /home/iparask/Virtualenv/RadicalDevel
radical.utils     : v0.45-2-g82050c5@devel
saga-python       : split-9-gf9b04d8@devel
radical.pilot     : split-66-ge67c4f3@devel
radical.analytics : v0.1-115-g31b5065@master
```

Zip file with all the logs:
[1342.zip](https://github.com/radical-cybertools/radical.pilot/files/979801/1342.zip)

**Comment by andre-merzky** [comment]
Thanks for the ticket!  Looks like the state updates from the client side to the DB get stuck: its trying to push a couple K units after scheduling, but things get interrupted after 4 min of inactivity...

**Comment by iparask** [comment]
Hello Andre,

I am using the database you proposed on our call last week. Apparently, mine does not work as it should. The agent pulls all the units (100 in total), but it does not execute them all at once. It executes a few, waits for some time and then executes another batch. Does the ```db_poll_sleeptime``` affect the speed it pushes to the db or is there some kind of blocking command?

**Comment by iparask** [comment]
Update:
I set up the stall to 1000 units and I hit a limit on my DB to 500. Yours DB did not hit the limit, but it is extremely slow.

**Comment by ibethune** [comment]
Hi @andre-merzky, do we understand the issue well enough to say if this is something which is a regression since 0.45 (in which case we must fix), or it can be documented and/or deferred to a future release?

**Comment by andre-merzky** [comment]
I have no idea on this one, really.  I mean, if the MongoDB limits collection size to 500 (or whatever), then there is not much to be done, really.  But I don't see how/why our default DB should be suddenly slow.  Well, it will always be slower than a local DB, naturally, and we know that our 'protocol' to the DB does not deserve the name and does not behave nicely with many units - but we should not see 'pull 100 units; sleep; pull 100 units; sleep'-cycles.  Also, this is not reported by others (and I don't see it), so I wonder what triggers the difference :/

**Comment by ibethune** [comment]
@iparask is this still causing an issue for you?  The initial report says that you changed from agent configuration parameters - this might explain why it's not seen by others.

**Comment by ibethune** [comment]
As discussed, this is not a documented configuration so is not a priority to fix before the release.

**Comment by andre-merzky** [comment]
Iannis, 

can you please do another run where you see things idling between 100 unit polls, and send me the client and agent logs, please, and also the script you are running?  Thanks!

**Comment by iparask** [comment]
Did I do this?

**Comment by andre-merzky** [comment]
interesting question :)  I don't think so.

**Comment by andre-merzky** [comment]
ping

**Comment by iparask** [comment]
Hello Andre, I just run this test with 100 units as a bulk pull. It run, but before saying that this is a success how can I verify it?

**Comment by iparask** [comment]
Okay! Checked with Manuel's help. I can say that this works! At least for Wrangler
